src/tyr/route_serializer_osrm.cc

```diff
@@ -1386,6 +1448,22 @@ json::ArrayPtr serialize_legs(const google::protobuf::RepeatedPtrField<valhalla:
       ++recost_itr;
     }
 
+    // Add admin country codes to leg json
+    auto admins = json::array({});
+    for (const auto& admin : path_leg.admin()) {
+      auto admin_map = json::map({});
+      if (admin.has_country_code()) {
+        admin_map->emplace("iso_3166_1", admin.country_code());
+        auto country_iso3 = iso2_to_iso3.find(admin.country_code());
+        if (country_iso3 != iso2_to_iso3.end()) {
+          admin_map->emplace("iso_3166_1_alpha3", country_iso3->second);
+        }
+      }
+      // TODO: iso_3166_2 state code
```

Comment:
So does this mean we will have different admin index pointing to the same country code?

Reply:
That's right yes, we'll have duplicate-looking entries in the admins list. I noticed this when crossing state boundaries in the US like this:
```
"admins: [{
  "iso_3166_1: "US",
  "iso_3166_1_alpha3: "USA"
}, {
  "iso_3166_1: "US",
  "iso_3166_1_alpha3: "USA"
}]
```

Other than looking strange in the JSON response, do you think this'll be a problem? I couldn't think of a straightforward workaround for this without changing how admins are listed and indexed.

Reply:
just thought it might look odd to some users - since we are not outputting the state level admin at this time; yeah, not worth changing if we output state level admin in the future